### PR TITLE
chore(): rm `fabric.filterBackend` => `getFilterBackend`

### PR DESCRIPTION
**Pull request by ShaMan123**
prepare for dropping fabric NS

## Motivation

Prepare for dropping fabric namespace

## Description

The ref `fabric.filterBakend` is gone, I wonder if the dev needs to set the backend for any reason because it is now impossible.
`getFilterBackend` is  a lazy initializer.

### CODE REVIEW

File: src/shapes/image.class.ts [hunk 1]
```diff
@@ -181,7 +181,7 @@ export class Image extends FabricObject {
    * Delete a single texture if in webgl mode
    */
   removeTexture(key: string) {
-    const backend = fabric.filterBackend;
+    const backend = getFilterBackend(false);
     if (backend && backend.evictCachesForKey) {
```
Comment (ShaMan123) [inline src/shapes/image.class.ts:185]: should we use instanceof WebGL?

File: src/filters/FilterBackend.ts [hunk 2]
```diff
@@ -16,15 +18,18 @@ export function initFilterBackend(): FilterBackend {
...
 /**
- * @todo refactor to a module w/o assigning to fabric
+ *
+ * @param [strict] pass `true` to create the backend if it wasn't created yet (default behavior),
+ * pass `false` to get the backend ref without mutating it
  */
-export function getFilterBackend(): FilterBackend {
-  if (!fabric.filterBackend) {
-    fabric.filterBackend = initFilterBackend();
+export function getFilterBackend(strict = true): FilterBackend {
```
[the remark this answers is not in the document]
Comment (ShaMan123) [reply src/filters/FilterBackend.ts:25]: `Image#removeTexture`: we don't want it to create the backend, only to probe it

[the remark this answers is not in the document]
Comment (asturur) [reply src/filters/FilterBackend.ts:25]: ok so is the issue that initElement calls removeTexture? and that we don't want to strictly couple Image with having a filter Backend?

[the remark this answers is not in the document]
Comment (ShaMan123) [reply src/filters/FilterBackend.ts:25]: Seems correct

**Comment by asturur** [comment]
Merging this and coming back with
- setFilterBackend ( when you want to swap imperatively from one to other )
- eventually getFilterBackend and getOrInitFilterBackend insted of strict because i really don't like strict = true
(cant see the strictly get it or not strictly get it meaning )

**Review by asturur — APPROVED** [review]

**Comment by ShaMan123** [comment]
> cant see the strictly get it or not strictly get it meaning

strict in terms of existence
But whatever. It is a matter of taste.

**Comment by ShaMan123** [comment]
Maybe it is only a naming issue.

**Pull request merged**